01_install_requirements.sh

```diff
@@ -65,3 +60,75 @@ if ! which dep 2>&1 >/dev/null ; then
     mkdir -p $GOPATH/bin
     curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
 fi
+
+# httpd-tools provides htpasswd utility
+sudo yum install -y httpd-tools
+
+sudo mkdir -pv ${REGISTRY_DIR}/{auth,certs,data}
+sudo chown -R $USER:$USER ${REGISTRY_DIR}
+
+pushd $REGISTRY_DIR/certs
+SSL_HOST_NAME="${LOCAL_REGISTRY_ADDRESS}"
+
+if [[ $( echo $SSL_HOST_NAME | grep -Eo '^[[:digit:]]{1,3}\.[[:digit:]]{1,3}\.[[:digit:]]{1,3}\.[[:digit:]]{1,3}$') ]];then
+    SSL_EXT_8="subjectAltName = IP:${SSL_HOST_NAME}"
+    SSL_EXT_7="subjectAltName = IP:${SSL_HOST_NAME}"
+else
+    SSL_EXT_8="subjectAltName = otherName:${SSL_HOST_NAME}"
+    SSL_EXT_7="subjectAltName = DNS:${SSL_HOST_NAME}"
+fi
+
+openssl genrsa -out ${REGISTRY_DIR}/certs/registry.key 2048
+
+if [ "${RHEL8}" = "True" ] ; then
+    openssl req -x509 \
+        -key ${REGISTRY_DIR}/certs/registry.key \
+        -out ${REGISTRY_DIR}/certs/registry.crt \
+        -days 365 \
+        -addext "${SSL_EXT_8}" \
+        -subj "/C=US/ST=NC/L=Raleigh/O=Test Company/OU=Testing/CN=${SSL_HOST_NAME}"
+else
+    SSL_TMP_CONF=$(mktemp 'my-ssl-conf.XXXXXX')
+    cat > ${SSL_TMP_CONF} <<EOF
+[req]
+distinguished_name = req_distinguished_name
+
+[req_distinguished_name]
+CN = ${SSL_HOST_NAME}
+
+[SAN]
+basicConstraints=CA:TRUE,pathlen:0
+${SSL_EXT_7}
+EOF
+
+    openssl req -x509 \
+        -key ${REGISTRY_DIR}/certs/registry.key \
+        -out  ${REGISTRY_DIR}/certs/registry.crt \
+        -days 365 \
+        -config ${SSL_TMP_CONF} \
+        -extensions SAN \
+        -subj "/C=US/ST=NC/L=Raleigh/O=Test Company/OU=Testing/CN=${SSL_HOST_NAME}"
+fi
+
+popd
+
+htpasswd -bBc ${REGISTRY_DIR}/auth/htpasswd ${REGISTRY_USER} ${REGISTRY_PASS}
+
+sudo cp ${REGISTRY_DIR}/certs/registry.crt /etc/pki/ca-trust/source/anchors/
+sudo update-ca-trust
+
+reg_state=$(sudo podman inspect registry --format  "{{.State.Status}}" || echo "error")
+if [[ "$reg_state" != "running" ]]; then
+    sudo podman rm registry -f || true
+
+    sudo podman run -d --name registry -p ${LOCAL_REGISTRY_PORT}:5000 \
+        -v ${REGISTRY_DIR}/data:/var/lib/registry:z \
+        -v ${REGISTRY_DIR}/auth:/auth:z \
+        -e "REGISTRY_AUTH=htpasswd" \
+        -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+        -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
```

Comment:
I wondering if we should include these at all, leaving them out turns off http auth, given that this is a dev tool it might make things simpler while debugging problems if their are no auth requirments.

Reply:
While debuging is simpler but having secret that references just local mirror ensures we don't _accidentally_ pull images from other sources.